File: skills/opencode-sdk/assets/hello-sdk.go
// Drive OpenCode from Go via the official SDK.
// Verify the surface in references/go.md and re-check github.com/sst/opencode-sdk-go for drift.
//
// Official module (NOT the anomalyco/manno23 forks):
//   go get -u 'github.com/sst/opencode-sdk-go@v0.19.2'   // requires Go 1.22+
//
// NewClient() targets a running `opencode serve` (default http://127.0.0.1:4096).
// Point it elsewhere or add auth with option.With* (e.g. option.WithBaseURL,
// option.WithHeader for OPENCODE_SERVER_PASSWORD basic auth).
package main

import (
	"context"
	"errors"
	"fmt"

	"github.com/sst/opencode-sdk-go"
)

func main() {
	client := opencode.NewClient()

	sessions, err := client.Session.List(context.TODO(), opencode.SessionListParams{})
	if err != nil {
		// All request params are wrapped in opencode.F(...) / Null[T]() / Raw[T]().
		var apiErr *opencode.Error
		if errors.As(err, &apiErr) {
			fmt.Println(string(apiErr.DumpRequest(true)))
		}
		panic(err.Error())
	}

	fmt.Printf("%+v\n", sessions)
}

File: skills/opencode-sdk/assets/hello-sdk.ts
// Drive OpenCode from JS/TS via @opencode-ai/sdk.
// Verify the surface in references/sdk.md and re-check https://opencode.ai/docs/sdk/ for drift.
//   npm install @opencode-ai/sdk
//
// createOpencode() spawns `opencode serve` AND returns a connected client.
// To attach to an already-running server instead, use:
//   import { createOpencodeClient } from "@opencode-ai/sdk"
//   const client = createOpencodeClient({ baseUrl: "http://localhost:4096" })
// There is NO createOpencodeServer in the official SDK — do not use it.

import { createOpencode } from "@opencode-ai/sdk"

const { client, server } = await createOpencode({ hostname: "127.0.0.1", port: 4096 })

try {
  const health = await client.global.health() // { healthy: true, version }
  console.log("server", server.url, health)

  const session = await client.session.create({ body: {} })
  const id = session.data.id

  // Structured output: model is forced through a json_schema StructuredOutput tool.
  const result = await client.session.prompt({
    path: { id },
    body: {
      parts: [{ type: "text", text: "Summarize this repo in one sentence." }],
      format: {
        type: "json_schema",
        schema: {
          type: "object",
          properties: { summary: { type: "string" } },
          required: ["summary"],
        },
      },
    },
  })

  if (result.data.info.error?.name === "StructuredOutputError") {
    console.error("structured output failed:", result.data.info.error.message)
  } else {
    console.log(result.data.info.structured_output)
  }
} finally {
  await server.close()
}
